### DISCUSSION TITLE
Support multiple storage options in persist-localstorage

### DISCUSSION THREAD

**Discussion by DaniAkash** [TanStack/query#1555] [Ideas]
I was checking out the source code for [persist-localstorage-experimental](https://github.com/tannerlinsley/react-query/blob/master/src/persist-localstorage-experimental/index.ts#L43) plugin & the implementation seems like easily extended to allow a different storage option other than localStorage. For eg: [localForage](https://github.com/localForage/localForage), [AsyncStorage](https://reactnative.dev/docs/asyncstorage) etc.

Is this something planned in the roadmap? If not I'll be happy to contribute towards implementing this...

**Comment by tannerlinsley** [comment]
I would be totally fine with making it situate storage agnostic and offering a few different adapters.

**Reply by DaniAkash** [reply]
@tannerlinsley I'll try creating a PR for this...
